### ISSUE TITLE
train params: `model` vs `pretrain`

### ISSUE THREAD

**Issue by Jordan-Pierce** [ultralytics/ultralytics#21166]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I'm assuming that if a user provides a string for `model` it's the same as if they had provided the same string for `pretrained`? The reason for having `pretrained` is if the user wanted to use the default model `yolov8` and train from scratch, which they would provide `False` for `pretrained`?

What then is the best way to transfer the encoder of a model previously trained on "similar" data, but not the same (different number of class categories, using `single_cls`, etc)?

I thought having `pretrained` be the path to the existing model, and set `resume` to `True` while pointing to a new dataset, but that resulted in overwriting existing weights.... :/

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Jordan-Pierce, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

---

This is an automated response 🤖. An Ultralytics engineer will assist you here soon!

**Comment by Y-T-G** [comment]
`pretrained` is the checkpoint from which the weights are transferred to the `model`. During training, a new model is built using the YAML of the `model` and the weights are transferred to that. If `pretrained` is `True` during training then the `model` is used as `pretrained`. If `pretrained` is a path to a checkpoint, then the weights are transferred from that checkpoint, ignoring the weights in the original `model`.

**Comment by Y-T-G** [comment]
> What then is the best way to transfer the encoder of a model previously trained on "similar" data, but not the same (different number of class categories, using `single_cls`, etc)?

You can use that checkpoint as `model`. The number of classes is automatically updated and the weights of affected layers are ignored while the rest are transferred.

**Comment by Jordan-Pierce** [comment]
Can `resume` be used at all for further fine-tuning, but on a new dataset? 

For example, once that previous model is done being trained and it's being used as the encoder for a *new* training session on a new dataset, the weights have already settled, and starting with the default training parameters would wash them out.

Are there ending parameters stored in the `.pt` that we can extract, to set the parameters values for a new session to simulate a `resume`?

**Comment by glenn-jocher** [comment]
The `resume` parameter is specifically designed to continue training on the same dataset from where it left off, not for fine-tuning on a new dataset. When you use `resume=True`, it restores the optimizer state, learning rate scheduler, and epoch number from the checkpoint, expecting to continue the exact same training session.

For fine-tuning a previously trained model on a new dataset, you should start a fresh training session using the trained model as your `model` parameter. While the checkpoint does contain training parameters like optimizer state and learning rate, these aren't directly accessible for transfer to a new training session on different data.

For effective fine-tuning, consider using a lower learning rate than the default (e.g., `lr0=0.001` or lower) to preserve the learned features while adapting to your new dataset. You can also adjust `warmup_epochs` and other parameters manually based on your fine-tuning requirements.

**Comment by Y-T-G** [comment]
You can check this

https://www.reddit.com/r/Ultralytics/comments/1eolwl8/the_correct_way_to_train_from_a_previously/

**Comment by Jordan-Pierce** [comment]
Great this actually helped a lot, much appreciated @glenn-jocher @Y-T-G 

Last question: what's the best way to go about using the encoder trained for image classification (imagine it's trained on all the bounding box crops) on a new model used for object detection? Same thing?

Additionally, is it possible to use the image classifier encoder as the encoder for this^ object detection model, but set the `cls` to `0.0` so that the detection model uses the encoder's classifications, and only learns bounding boxes (not their labels)?

**Comment by Y-T-G** [comment]
> Last question: what's the best way to go about using the encoder trained for image classification (imagine it's trained on all the bounding box crops) on a new model used for object detection? Same thing?

You can provide the classification model as `pretrained`.

> Additionally, is it possible to use the image classifier encoder as the encoder for this^ object detection model, but set the `cls` to `0.0` so that the detection model uses the encoder's classifications, and only learns bounding boxes (not their labels)?

Setting `cls` to 0 would probably result in model losing the classification performance as the weights get updated. You can try freezing the classification branch 

https://github.com/ultralytics/ultralytics/issues/19826#issuecomment-2746102024

`22.cv3` is classification branch.

**Comment by Jordan-Pierce** [comment]
@Y-T-G so something along the lines of:

```python
# Load the detection model to start with, then use the pretrained weights for the encoder
model = YOLO(model="best_detection.pt", pretrained="best_classification.pt")

# Freeze the encoder
freeze = [f"{k}" for k in range(22)]
freeze += ["22.cv3"]

# Train the model
results = model.train(data="training_data.yaml", freeze=freeze, task='detect', lr0=0.003, warmup_epochs=0, optimizer='AdamW' epochs=5)

**Comment by Y-T-G** [comment]
Yes

**Comment by kodaigit** [comment]
I would like to do double check whether this way is safe for fine tuning. 

When you finish training a model and get "model_1.pt",  and then if you want to fine-tune it on a new dataset, you can do ..

```
model = YOLO("model_1.pt")

model.train(some necessary params here)
```
Then, the training can be done using the weights from the checkpoint, but no optimizer and learning rates are transferred to the new training. Basically, what you provide in the `train()` will be used. 

Am I understanding correctly?

**Comment by Y-T-G** [comment]
Yes
